src/main/java/org/influxdb/dto/Point.java

```diff
@@ -196,6 +225,104 @@ public boolean hasFields() {
       return !fields.isEmpty();
     }
 
+    /**
+     * Adds field map from object by reflection using {@link org.influxdb.annotation.Column}
+     * annotation.
+     *
+     * @param pojo POJO Object with annotation {@link org.influxdb.annotation.Column} on fields
+     * @return the Builder instance
+     */
+    public Builder addFieldsFromPOJO(final Object pojo) {
+
+      Class<? extends Object> clazz = pojo.getClass();
+
+      for (Field field : clazz.getDeclaredFields()) {
+
+        Column column = field.getAnnotation(Column.class);
+
+        if (column == null) {
+          continue;
+        }
+
+        String fieldName = column.name();
+
+        if (isPublic(field)) {
+          addFieldByAttribute(pojo, field, column, fieldName);
+        } else {
+          addFieldByMethod(pojo, clazz, field, column, fieldName);
+        }
+      }
+
+      if (this.fields.isEmpty()) {
+        throw new BuilderException("Class " + pojo.getClass().getName()
+            + " has no @" + Column.class.getSimpleName() + " annotation");
+      }
+
+      return this;
+    }
+
+    private boolean isPublic(final Field field) {
+      int modifiers = field.getModifiers();
+      return Modifier.isPublic(modifiers);
+    }
+
+    private void addFieldByMethod(final Object pojo, final Class<? extends Object> clazz,
```

Comment:
If I'm not wrong, this implements a different behavior from the initial toPOJO implementation where we get data directly from the field instead of unnecessarily calling the accessor method. Is there any good reason to have this call?